UI Improvements to Coref Demo

aaronsarnat opened the pull request:
![image](https://user-images.githubusercontent.com/8367927/44602601-eabc2f80-a794-11e8-9306-cf227e6b219c.png)

---

This PR
---

- Updates Coref demo experience to use new `<Highlight />` component
- Improves `<Highlight />` component

aaronsarnat commented:
@schmmd assigning to you for review. FYI I have some minor CSS tweaks I still want to make before I merge this (esp. relating to nested refs), but I figured I could have you start looking at the logic while I make small edits to the stylesheet.

schmmd commented on demo/src/components/CorefComponent.js:
```diff
@@ -116,62 +116,133 @@ class CorefOutput extends React.Component {
+      const { activeIds, isClicking, selectedId } = this.state;
+      const { tokens, clusters } = this.props;
+
+      // Span tree data transform code courtesy of Michael S.
```
Consider putting this code into a helper function that just transforms the JSON structure.  That would make it easier to test if we find any problems.

aaronsarnat replied:
good idea. i'll take a look at that

aaronsarnat commented:
@schmmd PTAL. I feel good about the state this PR is in now and am ready to merge, but thought it was worth getting another look at it from you (or whoever) since I made a number of changes since you approved this:

- Fixed nested highlight interaction bug where the wrong cluster was being selected
- Refactored highlight component structure to support cleaner-looking layout (side-aligned labels now fill height of container, things aren't quite as scrunched, and the new structure should be more robust when dealing with edge cases) 
- Modularized the `<Highlight />` component a bit more
- Browser tested on the demo's most used browsers

---

Here's a screen video of the latest (sped up a little):

![highlight-ux](https://user-images.githubusercontent.com/8367927/44644207-24a15780-a988-11e8-9060-0e12fbaeea4b.gif)
